File: app/src/main/java/com/ruideraj/secretelephant/match/MatchExchange.java
package com.ruideraj.secretelephant.match;

import android.os.Parcel;
import android.os.Parcelable;

import com.ruideraj.secretelephant.contacts.Contact;

import java.util.ArrayList;

/**
 * Class containing all the data needed for a gift exchange.
 */
public class MatchExchange implements Parcelable {

    private ArrayList<Contact> mContacts;
    private int[] mMatches;
    private int mMode;

    public MatchExchange(ArrayList<Contact> contacts, int[] matches, int mode) {
        mContacts = contacts;
        mMatches = matches;
        mMode = mode;
    }

    public ArrayList<Contact> getContacts() {
        return mContacts;
    }

    public void setContacts(ArrayList<Contact> contacts) {
        this.mContacts = contacts;
    }

    public int[] getMatches() {
        return mMatches;
    }

    public void setMatches(int[] matches) {
        this.mMatches = matches;
    }

    public int getMode() {
        return mMode;
    }

    public void setMode(int mode) {
        this.mMode = mode;
    }


    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeTypedList(this.mContacts);
        dest.writeIntArray(this.mMatches);
        dest.writeInt(this.mMode);
    }

    public MatchExchange() {
    }

    protected MatchExchange(Parcel in) {
        this.mContacts = in.createTypedArrayList(Contact.CREATOR);
        this.mMatches = in.createIntArray();
        this.mMode = in.readInt();
    }

    public static final Parcelable.Creator<MatchExchange> CREATOR = new Parcelable.Creator<MatchExchange>() {
        @Override
        public MatchExchange createFromParcel(Parcel source) {
            return new MatchExchange(source);
        }

        @Override
        public MatchExchange[] newArray(int size) {
            return new MatchExchange[size];
        }
    };
}

File: app/src/main/java/com/ruideraj/secretelephant/match/MatchExchange.kt
package com.ruideraj.secretelephant.match

import android.os.Parcelable
import com.ruideraj.secretelephant.Mode
import com.ruideraj.secretelephant.contacts.Contact
import kotlinx.parcelize.Parcelize

@Parcelize
data class MatchExchange(val contacts: List<Contact>, val matches: IntArray, val mode: Mode)
    : Parcelable {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as MatchExchange

        if (contacts != other.contacts) return false
        if (!matches.contentEquals(other.matches)) return false
        if (mode != other.mode) return false

        return true
    }

    override fun hashCode(): Int {
        var result = contacts.hashCode()
        result = 31 * result + matches.contentHashCode()
        result = 31 * result + mode.hashCode()
        return result
    }


}